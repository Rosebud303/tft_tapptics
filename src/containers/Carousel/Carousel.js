import React, { Component } from 'react';
import InitialCharacter from '../../components/InitialCharacter/InitialCharacter';
import { connect } from 'react-redux'

export class Carousel extends Component {
  constructor(props) {
    super(props);
    
  }
  
  render() {
    const filteredHeroes = this.props.champions.filter( champion => {
      return champion.attributes.data.cost <= 2
    })
    const initialHeroes = filteredHeroes.map(hero => <InitialCharacter key={hero.attributes.data.name} 
                                                                       name={hero.attributes.data.name}
                                                                       img={hero.attributes.data.champion_thumbnail}
                                                                       synergy={hero.attributes.origin_class_type.data[0].attributes.data.name} />)
    return (
      <div className='full-car'>
        <h1 className='car-title'>Choose A Champion</h1>
        <div className='carousel'>
          {initialHeroes}
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  champions: state.champions
})

export default connect(mapStateToProps)(Carousel)
