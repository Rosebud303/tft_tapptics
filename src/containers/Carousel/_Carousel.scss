.carousel {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.full-car {
  width: 100%;
  height: 100vh;
  background: linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.9) ), url('../../assets/images/tft.jpg');
  background-repeat: no-repeat;
  background-position: center; 
  background-size: cover;
  box-shadow: 10px 10px 3px #3360986e;
  border: 3px solid black;
  text-align: center;
}

.car-title {
  display: block;
}

.images {
  width: 15%;
  background: red;
}

.car-imgs {
  width:100%;
  height: 100%;
}