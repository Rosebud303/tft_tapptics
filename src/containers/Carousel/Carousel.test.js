import React from 'react';
import { shallow } from 'enzyme';
import Carousel from './Carousel';

describe('Champions', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Carousel />);
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});