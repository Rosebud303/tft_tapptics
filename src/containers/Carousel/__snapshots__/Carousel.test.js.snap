// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Champions should match the snapshot with all the data passed in 1`] = `
<div
  className="full-car"
>
  <h1
    className="car-title"
  >
    Choose A Champion
  </h1>
  <div
    className="carousel"
  >
    <div
      className="images"
    >
      <a
        data-for="this-one"
        data-tip={true}
      >
        <img
          className="car-imgs"
          src="https://i.imgur.com/9YG3Clp.png"
        />
      </a>
      <ReactTooltip
        clickable={false}
        id="this-one"
        insecure={true}
        resizeHide={true}
        type="success"
        wrapper="div"
      >
        <span>
          IT WORKS SON!
        </span>
      </ReactTooltip>
    </div>
    <div
      className="images"
    >
      <a
        data-for="thistwo"
        data-tip={true}
      >
        <img
          className="car-imgs"
          src="https://i.imgur.com/DMOVd7b.png"
        />
      </a>
      <ReactTooltip
        clickable={false}
        id="thistwo"
        insecure={true}
        resizeHide={true}
        type="success"
        wrapper="div"
      >
        <span>
          DUUUUDE!
        </span>
      </ReactTooltip>
    </div>
    <div
      className="images"
    >
      <a
        data-for="this-three"
        data-tip={true}
      >
        <img
          className="car-imgs"
          src="https://i.imgur.com/0keMK4n.png"
        />
      </a>
      <ReactTooltip
        clickable={false}
        id="this-three"
        insecure={true}
        resizeHide={true}
        type="success"
        wrapper="div"
      >
        <span>
          DOG IM TELLIN YOU!
        </span>
      </ReactTooltip>
    </div>
    <div
      className="images"
    >
      <a
        data-for="this-4"
        data-tip={true}
      >
        <img
          className="car-imgs"
          src="https://i.imgur.com/e08Gynb.png"
        />
      </a>
      <ReactTooltip
        clickable={false}
        id="this-4"
        insecure={true}
        resizeHide={true}
        type="success"
        wrapper="div"
      >
        <span>
          BELIEVE ME
        </span>
      </ReactTooltip>
    </div>
    <div
      className="images"
    >
      <a
        data-for="this-5"
        data-tip={true}
      >
        <img
          className="car-imgs"
          src="https://i.imgur.com/OdC7Spy.png"
        />
      </a>
      <ReactTooltip
        clickable={false}
        id="this-5"
        insecure={true}
        resizeHide={true}
        type="success"
        wrapper="div"
      >
        <span>
          BELIEVE ME YET!??
        </span>
      </ReactTooltip>
    </div>
  </div>
</div>
`;
